specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json

```diff
@@ -1261,6 +1339,76 @@
         "aggregation"
       ]
     },
+    "CategoricalVariable": {
+      "description": "Categorical variable represents signal that needs to be analyzed based on the count or duration of occurrence of limited set of defined values.",
+      "x-ms-discriminator-value": "categorical",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Variable"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "value": {
+              "description": "Value time series expression is used to represent the value of the signal that is going to be categorized. It can evaluate to only 'String' or 'Long' type for categorical variables.",
+              "$ref": "#/definitions/Tsx"
+            },
+            "interpolation": {
+              "description": "Categorical variable supports only 'step' interpolation.",
+              "$ref": "#/definitions/Interpolation"
+            },
+            "categories": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/TimeSeriesAggregateCategory"
+              }
+            },
+            "defaultCategory": {
+              "$ref": "#/definitions/TimeSeriesDefaultCategory"
+            }
+          }
+        }
+      ],
+      "required": [
+        "kind",
+        "value",
+        "defaultCategory"
+      ]
+    },
+    "TimeSeriesAggregateCategory": {
+      "description": "Category used in categorical variables. A category is defined by 'label' and the 'values' that are assigned this label.",
+      "type": "object",
+      "properties": {
+        "label": {
+          "description": "The name of the category which will be used in constructing the output variable names.",
+          "type": "string"
+        },
+        "values": {
+          "description": "The list of values that a category maps to. Can be either a unique list of string or list of long.",
+          "type": "array",
+          "items": {
+            "type": "object"
```

Comment:
The description and swagger are in disagreement as to what the types can be here. A `string` or `integer` in swagger/openapi are **not** `object`s.